packages/i18next/src/lib/InternationalizationHandler.ts

```diff
@@ -0,0 +1,203 @@
+	 */
+	public languagesLoaded = false;
+
+	/**
+	 * A `Set` of initially loaded namespaces.
+	 * @since 1.2.0
+	 */
+	public namespaces = new Set<string>();
+
+	/**
+	 * A `Map` of `i18next` language functions keyed by their language code.
+	 * @since 1.0.0
+	 */
+	public readonly languages = new Map<string, TFunction>();
+
+	/**
+	 * The options InternationalizationHandler was initialized with in the client.
+	 * @since 1.0.0
+	 */
+	public readonly options: InternationalizationOptions;
+
+	/**
+	 * The director passed to `i18next-fs-backend`.
+	 * Also used in {@link InternationalizationHandler.walkLanguageDirectory}.
+	 * @since 1.2.0
+	 */
+	public readonly languagesDirectory: string;
+
+	/**
+	 * The backend options for `i18next-fs-backend` used by `i18next`.
+	 * @since 1.0.0
+	 */
+	protected readonly backendOptions: i18nextFsBackend.i18nextFsBackendOptions;
+
+	/**
+	 * @param options The options that `i18next`, `i18next-fs-backend`, and {@link InternationalizationHandler} should use.
+	 * @since 1.0.0
+	 * @constructor
+	 */
+	public constructor(options?: InternationalizationOptions) {
+		this.options = options ?? {};
+		this.languagesDirectory = this.options.defaultLanguageDirectory ?? join(getRootData().root, 'languages');
+
+		this.backendOptions = {
+			loadPath: join(this.languagesDirectory, '{{lng}}', '{{ns}}.json'),
+			addPath: this.languagesDirectory,
+			...this.options.backend
+		};
+	}
+
+	/**
+	 * The method to be overridden by the developer.
+	 *
+	 * @note In the event that fetchLanguage is not defined or returns null / undefined, the defaulting from {@link fetchLanguage} will be used.
+	 * @since 2.0.0
+	 * @return A string for the desired language or null for no match.
+	 * @see {@link fetchLanguage}
+	 * @example
+	 * ```typescript
+	 * // Always use the same language (no per-guild configuration):
+	 * client.fetchLanguage = () => 'en-US';
+	 * ```
+	 * @example
+	 * ```typescript
+	 * // Retrieving the language from an SQL database:
+	 * client.fetchLanguage = async (context) => {
+	 *   const guild = await driver.getOne('SELECT language FROM public.guild WHERE id = $1', [context.guild.id]);
+	 *   return guild?.language ?? 'en-US';
+	 * };
+	 * ```
+	 * @example
+	 * ```typescript
+	 * // Retrieving the language from an ORM:
+	 * client.fetchLanguage = async (context) => {
+	 *   const guild = await driver.getRepository(GuildEntity).findOne({ id: context.guild.id });
+	 *   return guild?.language ?? 'en-US';
+	 * };
+	 * ```
+	 * @example
+	 * ```typescript
+	 * // Retrieving the language on a per channel basis, e.g. per user or guild channel (ORM example but same principles apply):
+	 * client.fetchLanguage = async (context) => {
+	 *   const channel = await driver.getRepository(ChannelEntity).findOne({ id: context.channel.id });
+	 *   return channel?.language ?? 'en-US';
+	 * };
+	 * ```
+	 */
+	public fetchLanguage: (context: InternationalizationContext) => Promise<string | null> | string | null = () => null;
```

Comment:
Don't we have Awaitable?

Reply:
`Awaited` yes, and i18next already depends on utilities. Please add this @kyranet